### PR TITLE
Make lockfile expire after function timeout

### PR DESCRIPTION
**Pull request by andrewbranch**
This prevents Azure Functions from thinking there’s still an instance running if an old instance got unceremoniously terminated without being able to clean up after itself by assuming that an instance cannot still be running if it claimed to have started longer ago than the function timeout would allow. I’ve seen this happen exactly one time since moving to Azure Functions, so I think we can expect it to happen every once in a while, I guess.

### CODE REVIEW

File: packages/publisher/src/main.ts [hunk 1]
```diff
@@ -44,28 +44,32 @@ export default function main() {
...
-      }
-    );
+  await writeFile(lockFilePath, JSON.stringify({ timestamp: currentTimeStamp() }));
+  cb().then(
+    () => remove(lockFilePath),
+    async error => {
+      console.error(error?.stack || error?.message || error);
+      applicationinsights.defaultClient.trackException({
+        exception: error,
+      });
```
Comment (andrewbranch) [inline packages/publisher/src/main.ts:63]: Ignore whitespace to review

File: packages/publisher/tsconfig.json [hunk 2]
```diff
@@ -2,11 +2,7 @@
     "extends": "../../tsconfig.base.json",
     "compilerOptions": {
         "outDir": "dist",
-        "rootDir": "src",
-        "baseUrl": ".",
-        "paths": {
-            "*": ["src/types/*"]
-        }
+        "rootDir": "src"
```
Comment (andrewbranch) [inline packages/publisher/tsconfig.json:5]: Drive by clean up 🚗

File: packages/utils/src/io.ts [hunk 3]
```diff
@@ -39,19 +39,29 @@ export function readFileSync(path: string): string {
...
   try {
-    return await readJson(filePath);
+    return await readJson(filePath, isObject);
   } catch (e) {
     console.error(`Run ${generatedBy} first!`);
     throw e;
   }
 }
 
-export function readJsonSync(path: string): object {
```
Comment (andrewbranch) [inline packages/utils/src/io.ts:49]: Pro tip, `JSON.parse` does not always produce an `object` 🤦 

Quality of life enhancements throughout this section

**Review by sandersn — APPROVED** [review]
Looks reasonable enough to try shipping next week!

**Pull request merged**